map_model/src/pathfind/mod.rs

```diff
@@ -675,6 +677,29 @@ pub fn zone_cost(turn: &Turn, constraints: PathConstraints, map: &Map) -> Durati
     }
 }
 
+/// Heavily penalize crossing into an access-restricted zone that doesn't allow this mode.
+pub fn zone_cost_v2(mvmnt: MovementID, constraints: PathConstraints, map: &Map) -> Duration {
```

Comment:
This is a copy of `zone_cost`. It'll get deduped when all the pathfinding v2 stuff is done.